flow/TDMetric.cpp

```diff
@@ -229,3 +235,143 @@ std::string MetricData::toString() const {
 	              rollTime,
 	              writer.getLength());
 }
+
+std::string createStatsdMessage(const std::string& name, StatsDMetric type, const std::string& val) {
+	return createStatsdMessage(name, type, val, {});
+}
+
+std::string createStatsdMessage(const std::string& name,
+                                StatsDMetric type,
+                                const std::string& val,
+                                const std::vector<std::pair<std::string, std::string>>& tags) {
+	ASSERT(!name.empty());
+	std::string msg = name + ":" + val;
+	switch (type) {
+	case StatsDMetric::GAUGE:
+		msg += "|g";
+		break;
+
+	case StatsDMetric::COUNTER:
+		msg += "|c";
+		break;
+	}
+
+	if (!tags.empty()) {
+		msg += "|#";
+		for (size_t i = 0; i < tags.size(); i++) {
+			msg = msg + tags[i].first + ":" + tags[i].second;
+			// If we know there is another tag coming, we should add a comma in the message
+			if (i != tags.size() - 1) {
+				msg += ",";
+			}
+		}
+	}
+
+	return msg;
+}
+
+MetricsDataModel knobToMetricModel(const std::string& knob) {
+	if (knob == "statsd") {
+		return MetricsDataModel::STATSD;
+	} else if (knob == "otel") {
+		return MetricsDataModel::OTEL;
+	}
```

Comment:
This should return `NONE` only if the knob is explicitly set to `none`, otherwise it should throw an error. Imagine a scenario where the user typos and sets the knob to `oteI` (that's a capital i, not a lowercase L). It would silently set the data model to `NONE`, causing confusion.